src/CloudEvent.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+/**
+ * Copyright 2019 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\CloudFunctions;
+
+class CloudEvent
+{
+    // Required Fields
+    private $id;
+    private $source;
+    private $specversion;
+    private $type;
+
+    // Optional Fields
+    private $datacontenttype;
+    private $dataschema;
+    private $subject;
+    private $time;
+    private $data;
```

Comment:
This doesn't have to happen in this PR, but we should change `Context` to follow the same paradigm of private properties with getters

Reply:
A different PR sounds like it's in scope for BackgroundFunction changes.